test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +41,412 @@ public void LocalSources_PackageReader_NotNull()
             PackageReaderBase reader = func();
             Assert.IsType(typeof(PackageArchiveReader), reader);
         }
+
+        [Fact]
+        public async Task IconUrl_WithMalformedUrlScheme_ReturnsDefault()
+        {
+            var iconUrl = new Uri("httphttphttp://fake.test/image.png");
+
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+            VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+            Assert.Equal(expected: IconBitmapStatus.DefaultIcon, packageItemListViewModel.BitmapStatus);
+        }
+
+        [Fact]
+        public async Task IconUrl_WhenFileNotFound_ReturnsDefault()
+        {
+            var iconUrl = new Uri(@"C:\path\to\image.png");
+
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+            VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+            Assert.Equal(expected: IconBitmapStatus.DefaultIcon, packageItemListViewModel.BitmapStatus);
+        }
+
+        [Fact]
+        public async Task IconUrl_RelativeUri_ReturnsDefault()
+        {
+            // relative URIs are not supported in viewmodel.
+            var iconUrl = new Uri("resources/testpackageicon.png", UriKind.Relative);
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+            VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+            Assert.Equal(expected: IconBitmapStatus.DefaultIcon, packageItemListViewModel.BitmapStatus);
+        }
+
+        [Fact]
+        public async Task IconUrl_WithLocalPathAndColorProfile_LoadsImage()
+        {
+            var iconUrl = new Uri("resources/grayicc.png", UriKind.Relative);
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+            VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+
+            // TODO: this isn't working now...and i think it shouldn't work. follow up. (relative URIs...shouldn't work!!!)
```

Comment:
i think it will fail. need a run to finish first.
will leave this conversation open to track for now.

Reply:
figured out that the change was testing the fact that a new setting was put in the decoding routine...and making sure that it worked with that image.  so, since we don't support relative URIs...i just had to get the path to the assembly, find the image relative to that, and send in an absolute file uri to the viewmodel. 
fixed in latest iteration.